src/opsinputs/opsinputs_varobswriter_mod.F90

```diff
@@ -235,6 +236,13 @@ function opsinputs_varobswriter_create(self, f_conf, comm_is_valid, comm, channe
 allocate(self % channels(size(channels)))
 self % channels(:) = channels(:)
 
+call f_conf % get_or_die("NumVarChannels", self % NumVarChannels)
+allocate(self % varChannels(self % NumVarChannels))
```

Comment:
If these are allocated here won't it mean that localvarChannels are always allocated even if size zero and therefore the if allocated in the fill routine will always be use this.

Reply:
Yes the local will be allocated but then there are checks for if the size of varChannels > 0. 

I think I might have started doing the varChannels as a must include option and have later switched it to an optional. 

It might be an idea to simplify the logic that we use varChannels as the writout option, if it has size zero is reallocated to the channels and filled with the jopa channel values. Then we could remove one of the if blocks which is checking if varChannels is greater then size zero.